privacyidea/lib/smsprovider/FirebaseProvider.py

```diff
@@ -103,85 +107,88 @@ def submit_message(self, firebase_token, data):
         :return: bool
         """
         res = False
+        credentials = get_firebase_access_token(self.smsgateway.option_dict.get(FirebaseConfig.JSON_CONFIG))
 
-        credentials = get_firebase_access_token(self.smsgateway.option_dict.get(
-            FIREBASE_CONFIG.JSON_CONFIG))
-
-        authed_session = AuthorizedSession(credentials)
+        proxies = {}
+        auth_request = None
+        if self.smsgateway.option_dict.get(FirebaseConfig.HTTPS_PROXY):
+            proxies["https"] = self.smsgateway.option_dict.get(FirebaseConfig.HTTPS_PROXY)
+            session = Session()
+            session.proxies.update(proxies)
+            auth_request = requests.Request(session=session)
+
+        if auth_request:
```

Comment:
No need to check if `auth_request` is set, just pass it to `AuthorizedSession`, default is `None`